R/data_prep.R

```diff
@@ -338,22 +344,25 @@ set_exp_intervals <- function(data_list, mapping = default_drug_admin(), subjid_
       )
     ) %>%
     dplyr::ungroup()
-
-  check_names(data, cols, subjid_var)
-  check_date_type(data, c(col_list$start_var, col_list$end_var))
-
   interval_df <- data %>%
     dplyr::mutate(
       detail_var = paste(
         .data[[col_list$detail_var]], "-",
         .data[[col_list$dose_var]],
         .data[[col_list$dose_unit_var]]
-      )
+      ),
+      trt_var = .data[[col_list$trt_var]]
     ) %>%
     dplyr::select(
-      tidyselect::all_of(c(subjid_var, cols[1:2], "set_id", "exp_dose", "detail_var"))
+      tidyselect::all_of(c(subjid_var, cols[1:2], "set_id", "exp_dose", "detail_var", "trt_var"))
+    ) %>%
+    dplyr::mutate(
+      group = dplyr::if_else(
+        !is.na(.data[["trt_var"]]),
+        paste("Drug Administration:", .data[["trt_var"]]),
```

Comment:
```suggestion
        paste0(col_list$label, ": ", .data[["trt_var"]]),
```
We should use the label argument of drug admin instead of hard coding Durg Administration :)